Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,122 @@
+
+We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
+
+### Certificates ###
+First, prepare the certificates that will be used for nodes to communicate in the cluster. Create a `certs/` directory, then move & run `generate-certs.sh` passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface is. You can acquire this through `ifconfig` or through:
+
+    $ ip addr show dev eth0
+
+if you already know the interface name. Then:
+
+    ~/kube $ mkdir certs && cd certs
+    ~/kube/certs $ mv ../generate-certs.sh .
+    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
+
+### Allow Routing ###
+_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+
+Windows nodes will each grab a /24 from the cluster CIDR, so it's 192.168.0.0/16, the first Windows node will use 192.168.0.1/24 for its pods. Pass the first two octets of your clister CIDR to generate the routes:
+
+    ~/kube $ generate-routes.sh 192.168
+
+### Prepare Manifests & Addons ###
+In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
+
+    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+
+This will generate a set of YAML files. You should [re]move the Python script so that K8s doesn't mistake it for a manifest and cause problems.
+
+### Configure & Run Kubernetes ###
+Run the script `configure-kubectl.sh` passing your external IP again. This will create a configuration in `~/.kube/config` containing all of the certificate data and other parameters:
+
+    ~/kube $ ./configure-kubectl.sh 10.123.45.67
+
+Run the script `start-kubelet.sh`. You may need to run it with `sudo`. Then, in another shell, watch your working directory (`~/kube`) until the folder `kubelet/` appears. Immediately copy the Kubernetes configuration file to it:
+
+    ~/kube $ sudo cp ~/.kube/config kubelet/
+
+**TODO**: Why is this necessary? Shouldn't this be done automatically by one of the containers?
+
+In another terminal session, run the Kubeproxy script, passing your cluster CIDR (you may also need to run this one with `sudo`):
+
+    ~/kube $ ./kubeproxy.sh 192.168
+
+After a few minutes, you should see the following system state:
+
+  - Under `docker ps`, you should see worker and pod containers mirroring the configs in `manifest/` and in `addons/`.
+  - A call to `~/kube/bin/kubectl cluster-info` should show info for the Kubernetes master API server, plus the DNS and heapster addons.
+  - A new interface `cbr0` under `ifconfig`, with your chosen cluster CIDR.
+
+## Join a Windows Node ##

```

Comment:
I just realized that we didn't tell customers what is needed on the Windows node :) For example, this is assuming that they have docker...are there any other assumptions around the version/sku of Windows or other pre-reqs? Can we provide instructions for getting docker installed?